src/Entity/WSCall.php

```diff
@@ -97,7 +100,48 @@ public function getLanguagePlugin() {}
   /**
    * {@inheritdoc}
    */
-  public function call($type, $key = NULL, $replacement = [], $argument = [], $options = [], &$method = '') {}
+  public function call($method = NULL, $replacements = [], $data = NULL, $options = [], $key = NULL, $tokens = [])  {
+    // Build out the Cache ID based on the parameters passed.
+    $cid_array = array_merge($options, $this->getOptions(), $replacements, $tokens, array('data' => $data, 'key' => $key));
+    $cid = md5(serialize($cid_array));
+    if ($cache = \Drupal::cache('wsdata')->get($cid)) {
+      $cache_data = $cache->data;
+    }
+    else {
+      $conn = $this->getConnector();
+      $options = array_merge($options, $this->getOptions());
+
+      if ($method and !in_array($method, $conn->getMethods())) {
+        throw new WSDataInvalidMethodException(sprintf('Invalid method %s on connector type %s', $method, $this->wsserverInst->wsconnector));
+      }
+      elseif (isset($options['method']) and in_array($options['method'], $conn->getMethods())) {
+        $method = $options['method'];
+      }
+      else {
+        $methods = $conn->getMethods();
+        $method = reset($methods);
+      }
+
+      // Fetch the cache tags for this call and the server instance call.
+      $cache_tags = array_merge($this->wsserverInst->getCacheTags(), $this->getCacheTags());
```

Comment:
Where are the encoder and decoders getting applied to the data?

Reply:
Encoders I can't see anywhere where they were implemented? I figure they should be part of the WSCall entity in the Call function that way they can be applied globally ?

Decoders are applied on the WSCall addData & getData function.